DS18x20 / Add Alarm Search functionality / onewire.py

ioiogamboa opened the pull request:
I added an argument <cmd> to functions "_search_rom" and "scan" in order to define optional action. You can execute a Roms search or Alarm search. The default action is SEARCH_ROM.

ioiogamboa committed:
Update onewire.py

I added an argument <cmd> to functions "_search_rom" and "scan" in order to define optional action. You can execute a Roms search or Alarm search. The default action is SEARCH_ROM.

tve commented on drivers/onewire/onewire.py:
```diff
@@ -46,23 +47,25 @@ def select_rom(self, rom):
-
-    def _search_rom(self, l_rom, diff):
+    
+    # Search bus roms or alarm roms. Argument <cmd> shall take SERCH_ROM (default) or ALARM_SEARCH constants.
```
Typo: `SERCH_ROM`

ioiogamboa replied:
Tomorrow I'll try to finish to program Commands to Recall and Copy Tl,Th and config register between Scratchpad and EEPROM to avoid rewrite resolution and alarm each time start uC.

tve reviewed (commented):
Nice addition! I'm not seeing any real docs to update, except the short examples for the esp32/esp8266, sigh.

ioiogamboa committed:
onewire.py / Fix missing Colon in _search_rom function

With this change I could compile new firmare from branch

ioiogamboa committed:
Update ds18x20.py

ioiogamboa committed:
update 18xB20.py / def Alarm(self) / Add Alarm Function 

Return de roms device than are outside range  of alarm writed in scratchpad Th Tl

ioiogamboa committed:
ds18x20 / Scratchpad and EEPROM comands

Implement function that enable the Scratchpad write data from/to internal device EEPROM

ioiogamboa committed:
ds18x20 / add colon to def functions

Fix missing colon at end of def functions

ioiogamboa committed:
ds18x20 / Fix Select Rom

Select rom before cmd relative to EEPROM

ioiogamboa commented:
I added three functionality for DS18X20 Temperature Sensor, and i checked it in a Nodemcu microcontroller. I need check it with many devices in the bus:
1) Alarm Search
2) Copy Th, Tl, and Config Register from Scratchpad to EEPROM.
3) Recall Th, Tl and Config Register from EEPROM to Scratchpad (this is not real usefull because when device start at power on recall EEPROM automatically).
Two files are involved: onewire.py and ds18x20.py.
I'll test it next few days.
